zerver/views/webhooks/mention.py

```diff
@@ -0,0 +1,36 @@
+# Webhooks for external integrations.
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+from zerver.models import Client, UserProfile
+
+from django.http import HttpRequest, HttpResponse
+from six import text_type
+from typing import Dict, Any, Iterable, Optional
+
+@api_key_only_webhook_view('Mention')
+@has_request_variables
+def api_mention_webhook(request, user_profile, client,
+                        payload=REQ(argument_type='body'),
+                        stream=REQ(default='Mention'),
+                        topic=REQ(default='News')):
+    # type: (HttpRequest, UserProfile, Client, Dict[str, Iterable[Dict[str, Any]]], text_type, Optional[text_type]) -> HttpResponse
+
+    try:
+        title = payload["title"]
+        source = payload["source_name"]
+        source_url = payload["url"]
+        description = payload["description"]
+    except KeyError as e:
+        return json_error(_("Missing key {} in JSON").format(str(e)))
+
+    # construct the body of the message
+    body = '%s\n**[%s](%s)**:\n%s' % (source, title, source_url, description)
```

Comment:
I think we don't want the extra link, because (1) it's probably not what someone wants to click on (they're going to want to click on the specific article, not the homepage of the relevant site), and (2) I expect it to interact somewhat poorly with Zulip's upcoming open graph previews feature.  So probably worth changing it.